target/shared/cpuinfo.c

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <myst/tcall.h>
+#include <fcntl.h>
+#include <string.h>
+
+int myst_tcall_cpuinfo_size()
+{
+    int fd, nbytes, size = 0;
+    char buf[1024];
+
+    fd = open("/proc/cpuinfo", O_RDONLY);
```

Comment:
You can use `ftell` to get the file size quicker:
```c
fp = fopen("/proc/cpuinfo", "r");
fseek(fp, 0L, SEEK_END);
sz = ftell(fp);
```

Reply:
Tried this, didnt work. Digging further it looks like /proc files depending on the seq file implementation of lseek don't support `SEEK_END`.